### PR TITLE
Eff Srchball OK

### PR DESCRIPTION
**Pull request by hensldm**
The final one.

### CODE REVIEW

File: include/z64effect.h [hunk 1]
```diff
@@ -257,7 +257,6 @@ typedef struct {
     /* 0x18 */ u8 unk18; // Always 1?
 } EffectSsOverlay; // size = 0x1C
 
-//! TODO: Review reuse of vec/gfx/actor fields across all effects
```
Comment (hensldm) [inline include/z64effect.h:260]: Reviewed these:
vec: used to store vector information of different kinds like position and rotation
gfx: used to store dls and textures, and in 1 instance stores a collisionpoly (new to MM)
actor: used to store actors, and in 1 instance stores an s16* (usually the actors access struct members directly with no opportunity to cast to a temp, so a void* is unlikely.)

Comment (EllipticEllipsis) [reply include/z64effect.h:260]: Right, so we're happy that the srchball instance should stay as a bug? Do we want to add a warning for it?

File: src/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/z_eff_ss_solder_srch_ball.c [hunk 2]
```diff
@@ -1,27 +1,89 @@
...
+    this->velocity = initParams->velocity;
+    this->accel = initParams->accel;
+    this->update = EffectSsSolderSrchBall_Update;
+    if (!(initParams->flags & SOLDERSRCHBALL_INVISIBLE)) {
+        this->draw = EffectSsSolderSrchBall_Draw;
+    }
+    this->life = 10;
+    this->rgScale = initParams->scale;
+    this->rFlags = initParams->flags;
+    this->actor = (Actor*)initParams->playerDetected; // actor field was incorrectly used as a pointer to something else
```
Comment (EllipticEllipsis) [inline src/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/z_eff_ss_solder_srch_ball.c:36]: Does anything else expect to read this struct's actor field as an actor? If so, it's a bug.

File: src/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/z_eff_ss_solder_srch_ball.c [hunk 3]
```diff
@@ -1,27 +1,89 @@
...
+    return 1;
+}
+
+void EffectSsSolderSrchBall_Draw(PlayState* play, u32 index, EffectSs* this) {
+    s32 pad;
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+    f32 scale = this->rgScale / 100.0f;
+
+    func_8012C28C(gfxCtx);
+    func_8012C2DC(play->state.gfxCtx);
```
[the remark this answers is not in the document]
Comment (hensldm) [reply src/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/z_eff_ss_solder_srch_ball.c:46]: yep

File: include/z64effect.h (hunk 1 above)
Comment (hensldm) [reply include/z64effect.h:260]: Personally I would prefer just the bug tag.

**Review by AngheloAlf — APPROVED** [review]

File: include/z64effect.h (hunk 1 above)
Comment (EllipticEllipsis) [reply include/z64effect.h:260]: I'd like to know what others think, I think I'd prefer to have the warning myself, but I know @AngheloAlf prefers not, at least for now.

File: src/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/z_eff_ss_solder_srch_ball.c [hunk 4]
```diff
@@ -33,7 +33,10 @@ u32 EffectSsSolderSrchBall_Init(PlayState* play, u32 index, EffectSs* this, void
     this->life = 10;
     this->rgScale = initParams->scale;
     this->rFlags = initParams->flags;
-    this->actor = (Actor*)initParams->playerDetected; //! @bug actor field used to store an s16*
+
+    //! @bug actor field used to store an s16*
+    // This is purly cosmetic. Nothing external will ever read this as an Actor, so there is no unintended side-effects.
```
Comment (EllipticEllipsis) [inline src/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/z_eff_ss_solder_srch_ball.c:38]:
```suggestion
    // This is purly cosmetic. Nothing external will ever read this as an Actor, so there are no unintended side-effects.
```

**Review by EllipticEllipsis — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]
lgtm

**Pull request merged**